### ISSUE TITLE
JupyterLab: make installed packages (via pip, conda) persist between restarts

### ISSUE THREAD

**Issue by elisabettai** [ITISFoundation/osparc-issues#748]
**Describe the user role**
User have the flexibility to customize their JupyterLabs by installing extensions. However, this has to be re-done every time the study is opened

**Describe the goal**
Make the installed packages persistent

**Describe the benefit**
It will save a lot of user time. E.g. for some complex use-cases installing a conda environment takes many minutes

**Comment by elisabettai** [comment]
@pcrespov, this is a big piece of work, but it will cover one of the biggest limitations of our Jupyter services.
What are we missing parts from simcore to enable this?

**Comment by pcrespov** [comment]
> @pcrespov, this is a big piece of work, but it will cover one of the biggest limitations of our Jupyter services. What are we missing parts from simcore to enable this?

Yes, at first sight, that would require persisting the entire virtual-environment.  Other more clever approaches might exists though ... but we would need to do some research. I suggest that POs decide whether we try this or not.

**Comment by elisabettai** [comment]
Also better to wait to have saving for dy-sidecar stable. This will add stress on saving volume.

**Comment by elisabettai** [comment]
Duplicate of: https://github.com/ITISFoundation/osparc-issues/issues/441
